brave-kafka/src/main/java/smartthings/brave/kafka/consumers/TracingConsumerInterceptor.java

```diff
@@ -0,0 +1,129 @@
+
+/**
+ * Copyright 2016-2017 SmartThings
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package smartthings.brave.kafka.consumers;
+
+import brave.Tracer;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.config.ConfigException;
+import smartthings.brave.kafka.EnvelopeProtos;
+import zipkin.Constants;
+import zipkin.Endpoint;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TracingConsumerInterceptor<K> implements ConsumerInterceptor<K, byte[]> {
+
+  private Tracer tracer;
+  private SpanNameProvider<K> nameProvider;
+  private final Endpoint kafkaEndpoint;
+  private final static Logger logger = Logger.getLogger(TracingConsumerInterceptor.class.getName());
+
+  // TODO: how to inject dependencies without using config
+
+  public TracingConsumerInterceptor() {
+    kafkaEndpoint = Endpoint.builder().serviceName("Kafka").build();
+  }
+
+  private final TraceContext.Extractor<EnvelopeProtos.Envelope> extractor = new TraceContext.Extractor<EnvelopeProtos.Envelope>() {
```

Comment:
Might want to define a new class for this to make stack traces cleaner.  Anonymous classes tend to be harder to follow.  Since this doesn't appear to capture any context it should be fine to define it as another class.